hips/draw/tests/test_simple.py

```diff
@@ -53,5 +55,34 @@ def test_make_sky_image():
     assert_allclose(data[200, 994], 3717.10091363)
     assert_allclose(data[200, 995], 3402.55292158)
 
+get_hips_order_pars = [
+    dict(order=7, width=2000, height=1000, fov="3 deg"),
+    dict(order=4, width=1000, height=500, fov="10 deg"),
+    dict(order=3, width=500, height=250, fov="15 deg"),
+]
+
+
+@requires_hips_extra()
+@pytest.mark.parametrize('pars', get_hips_order_pars)
+def test_get_hips_order(pars):
+    full_path = get_hips_extra_file('datasets/samples/2MASS6XH/properties')
+    hips_survey = HipsSurveyProperties.read(filename=full_path)
+    geometry = WCSGeometry.create_simple(
+        skydir=SkyCoord(0, 0, unit='deg', frame='icrs'),
+        width=pars['width'], height=pars['height'], fov=pars['fov'],
+        coordsys='icrs', projection='AIT'
+    )
+    assert get_hips_order(geometry, hips_survey) == pars['order']
+
+get_hips_order_for_resolution_pars = [
+    dict(tile_width=512, resolution=12.32, order=3),
+    dict(tile_width=256, resolution=160.22, order=3),
```

Comment:
You should choose cases that lead to different `orders` and test the extremes, i.e. corner / edge cases of the algorithm in that function.
At the moment you only chose values that will result in `order=3`.